pallets/compliance-manager/src/lib.rs

```diff
@@ -197,16 +225,41 @@ type Identity<T> = identity::Module<T>;
 
 /// Asset compliance and it's evaluation result
 #[cfg_attr(feature = "std", derive(Serialize, Deserialize, Debug))]
-#[derive(codec::Encode, codec::Decode, Clone, PartialEq, Eq)]
+#[derive(Encode, Decode, Clone, PartialEq, Eq)]
 pub struct AssetComplianceResult {
     /// This flag indicates if asset compliance should be enforced
     pub paused: bool,
     /// List of compliance requirements.
     pub requirements: Vec<ComplianceRequirementResult>,
+    /// It is treated differently from other compliance requirements because
+    /// it doesn't successfully execute the transaction even if it succeed. As txn
+    /// also depends on the successful verification of one of the `requirement` set by
+    /// the asset issuer. But it can fail the txn if it gets failed independently from
+    /// the result of other requirements.
+    ///
+    /// List of implicit requirements.
+    pub implicit_requirements: Option<ComplianceRequirementResult>,
     // Final evaluation result of the asset compliance
     pub result: bool,
 }
 
+impl AssetComplianceResult {
+    pub fn get_implicit_requirements(
+        from_result: Option<ConditionResult>,
+        to_result: Option<ConditionResult>,
+    ) -> Option<ComplianceRequirementResult> {
+        from_result.zip(to_result).and_then(|(f, t)| {
+            let final_result = f.result && t.result;
+            Some(ComplianceRequirementResult {
+                sender_conditions: vec![f],
+                receiver_conditions: vec![t],
+                id: None,
+                result: final_result,
+            })
+        })
+    }
```

Comment:
None of this is required. We should just use a different format for reporting implicit requirements.

Reply:
same as above